### PR TITLE
Add "-S" and "--sslclean" description to the message printed by gpfdist --help

### PR DESCRIPTION
**Pull request by 50wu**

### CODE REVIEW

**Comment by adam8157** [comment]
Where should users place the "-S"?

**Comment by hlinnaka** [comment]
Why would anyone ever use that option? By the description, all it does is make things slower.

**Comment by lij55** [comment]
I don't know the detail reason for '-S' but there is requirement for that: https://github.com/greenplum-db/gpdb/issues/1645#issuecomment-276119788

This PR add the missing descriptions.

**Comment by hlinnaka** [comment]
Ok. I think the best steps here are:

0. Close this PR.
1. Figure out if --sslclean is actually needed anywhere anymore.
2. If it is still needed, figure out what exactly the problem is that it's working around.
3. Then figure out if we can fix whatever that problem is, properly, without resorting to a flag like --sslclean.
4. If that fails, then document that problem, and how the --sslclean helps with that, in the help message.

**Comment by lij55** [comment]
Mostly agreed, @hlinnaka 

Workaround is for AIX version of gpfdist and we don't have those environment now.
When the environment is ready, we will work on it.

As this flag is used by customer already, my suggestion is that we merge this PR first and revert it when we solve the ugly workaround.  Is it acceptable?

**Comment by adam8157** [comment]
the default of `--sslclean` is already `0` actually (which means no workaround). https://github.com/greenplum-db/gpdb/pull/2044

I think this PR doesn't hurt, we could figure that out once we have an AIX env.

File: gpMgmt/doc/gpfdist_help [hunk 1]
```diff
@@ -139,6 +140,15 @@ OPTIONS
  The root directory (/) cannot be specified as certificate_path. 
 
 
+--sslclean <wait_time>
+
+ When the utility is run with the --ssl option, sets the number of seconds
+ that the utility delays before closing an SSL session and cleaning up the
+ SSL resources after it completes writing data to or from a Greenplum Database
+ segment. The default value is 0, no delay. The maximum value is 500 seconds.
```
Comment (danielgustafsson) [inline gpMgmt/doc/gpfdist_help]: The maximum value is 299 seconds, the check is: `if ( (opt.sslclean < 0) || (opt.sslclean > 300) )`

File: src/bin/gpfdist/gpfdist.c [hunk 2]
```diff
@@ -483,33 +483,35 @@ static void usage_error(const char* msg, int print_usage)
...
-						"        -m maxlen  : max data row length expected, in bytes. default is 32768\n"
+						"        -d dir          : serve files under the specified directory,  default is '.'\n"
+						"        -l logfn        : log filename\n"
+						"        -t tm           : timeout in seconds \n"
+						"        -m maxlen       : max data row length expected, in bytes. default is 32768\n"
+						"        -S              : open the file for synchronous I/O with the O_SYNC flag\n"
 #ifdef USE_SSL
-						"        --ssl dir  : start HTTPS server. Use the certificates from the specified directory\n"
+						"        --ssl dir       : start HTTPS server. Use the certificates from the specified directory\n"
+						"        --sslclean time : define the time to wait until cleanup the SSL resources,default is 0\n"
```
Comment (danielgustafsson) [inline src/bin/gpfdist/gpfdist.c]: Missing a space after comma: "resources,default"

File: gpMgmt/doc/gpfdist_help (hunk 1 above)
Comment (adam8157) [inline gpMgmt/doc/gpfdist_help]: The maximum value is 300, *_^

Comment (danielgustafsson) [inline gpMgmt/doc/gpfdist_help]: Yes, I was clearly tired reading that =)

**Comment by hlinnaka** [comment]
Have you got hold of an AIX box yet?

**Comment by pivotal-mike** [comment]
@adam8157 @lij55 can we confirm if this is still an issue for AIX?

**Comment by adam8157** [comment]
Lopp @pf-qiu , the anchor of Clients team (owning gpfdist) and the main developer of the stuff on AIX.

**Comment by lij55** [comment]
In fact we still don't have proper environment to reproduce it and verity this issue. As this PR only changes the message description of its behavior, I suggest to merge it first. The mentioned parameters are required and used by customer.
Any objection?

**Comment by hlinnaka** [comment]
On 11/09/2017 10:42 AM, Jasper Li wrote:
> In fact we still don't have proper environment to reproduce it and verity this issue. As this PR only changes the message description of its behavior, I suggest to merge it first. The mentioned parameters are required and used by customer.
> Any objection?

Yes. This really ought to be investigated properly. If we don't have an 
AIX environment to test on, yet we claim to support AIX, then that's a 
problem. Please acquire one. We can't just hand over an option to 
customers, when we don't understand what it's supposed to do or why it's 
needed.

- Heikki

**Comment by lij55** [comment]
It is a legacy feature brought years ago and this PR only add the missing description for it.
At least this PR help to make code and doc consistent. If we need to investigate, should a new issue be better when we have required hardware?

**Comment by lij55** [comment]
https://github.com/greenplum-db/gpdb/issues/3831

**Comment by lij55** [comment]
Close this one and will remove it with https://github.com/greenplum-db/gpdb/issues/3831

**Pull request closed without merging**